src/circuits/joinsplit.tcc

```diff
@@ -104,8 +105,9 @@ public:
     // the verifier on-chain
     joinsplit_gadget(
         libsnark::protoboard<FieldT> &pb,
+        libsnark::pb_variable<FieldT> zero,
```

Comment:
I'd keep the declaration of the ZERO pb_variable "inside" of the circuit, just before the allocation on the protoboard a few lines below. ZERO represents a wire of the circuit (here we want to "create the wire" and then make this ZERO wire to the other sub-circuits called by the Joinsplit circuit so that we don't need to declare it and constrain it again in other subcircuits)

Reply:
I am not sure to understand why the following is written in joinsplit  then:
"  // TODO: Remove ZERO and pass it in the constructor
        pb_variable<FieldT> ZERO; "

Reply:
Oh, this one may be a dummy one then. I don't know why I wrote it in here. Sorry for that. We want to pass &ZERO to the subcircuits called from joinsplit, but let's stick to ZERO  being an attribute of the joinsplit circuit and initialized and allocated in here.

Reply:
It was in the master branch, it disappeared indeed.
Sure no problem, will fix that!

Reply:
thanks